modules/juce_audio_devices/native/juce_win32_WASAPI.cpp

```diff
@@ -1297,6 +1303,31 @@ class WASAPIAudioIODevice  : public AudioIODevice,
                 triggerAsyncUpdate();
                 break; // Quit the thread... will restart it later!
             }
+            else
+            {
+checkAudioSessionDisconnect:

```

Comment:
Using goto like this makes me a bit uneasy about readability of the control flow. But I guess this is warranted for keeping the changes to Juce minimal.

Reply:
The other option I could think of was also a bit convoluted: break here, then keep the original check for sample rate changed, and finally after the loop has exited check again for disconnection. But at least that could catch cases where there's timeout for some other reason than disconnecting the device.